File: evidence/generated-systems/angular/insurance-on-angular/src/app/models/Application.js

// Define collection and schema for Application
export interface Application {
    applicationNumber:
	type : string
    submissionDate:
	type : Date
    Customer:
	type : Schema.Types.ObjectId
    Product:
	type : Schema.Types.ObjectId
    Distributor:
	type : Schema.Types.ObjectId
    Quotes:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Quote' }]
    SelectedQuote:
	type : Schema.Types.ObjectId
    Status:
 	type : String
#
    collection: 'applications'
}

File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/Application.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Application Declaration
//==============================================================
type Application struct {
    gorm.Model
     ApplicationNumber                                    string
    SubmissionDate                                                            time.Time
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
    ProductId         *uint
    Product           *InsuranceProduct `gorm:"foreignKey:ProductId"`
    DistributorId         *uint
    Distributor           *Distributor `gorm:"foreignKey:DistributorId"`
     Quotes           []Quote `gorm:"foreignKey:QuotesFromApplicationId"`
    SelectedQuoteId         *uint
    SelectedQuote           *Quote `gorm:"foreignKey:SelectedQuoteId"`
    Status                      ApplicationStatus

// parent associations as their child

}

